File: EggIncognito.CodeGen/Templates/java/src/test/java/com/egginc/mock/ServerTest.java
package com.egginc.mock;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.io.TempDir;
import java.io.IOException;
import java.nio.file.*;
import static org.junit.jupiter.api.Assertions.*;

class ServerTest {

    private static final String EID_PROTO = "MhJFSTAwMDAwMDAwMDAwMDAwMDE=";

    @Test
    void extractEidReturnsCorrectEid() {
        assertEquals("EI0000000000000001", Server.extractEid(EID_PROTO));
    }

    @Test
    void extractEidEmptyInput() {
        assertEquals("", Server.extractEid(""));
    }

    @Test
    void loadFixtureDefault(@TempDir Path dir) throws IOException {
        Path defDir = dir.resolve("default");
        Files.createDirectories(defDir);
        Files.write(defDir.resolve("test.binpb"), new byte[]{0x01, 0x02});

        String orig = Server.FIXTURES_PATH;
        Server.FIXTURES_PATH = dir.toString();
        try {
            byte[] data = Server.loadFixture("test", "");
            assertArrayEquals(new byte[]{0x01, 0x02}, data);
        } finally {
            Server.FIXTURES_PATH = orig;
        }
    }

    @Test
    void loadFixtureEidOverride(@TempDir Path dir) throws IOException {
        Files.createDirectories(dir.resolve("default"));
        Files.createDirectories(dir.resolve("eids/EI0000000000000001"));
        Files.write(dir.resolve("default/test.binpb"), new byte[]{0x01});
        Files.write(dir.resolve("eids/EI0000000000000001/test.binpb"), new byte[]{0x02});

        String orig = Server.FIXTURES_PATH;
        Server.FIXTURES_PATH = dir.toString();
        try {
            byte[] data = Server.loadFixture("test", "EI0000000000000001");
            assertArrayEquals(new byte[]{0x02}, data);
        } finally {
            Server.FIXTURES_PATH = orig;
        }
    }

    @Test
    void loadFixtureMissing(@TempDir Path dir) {
        String orig = Server.FIXTURES_PATH;
        Server.FIXTURES_PATH = dir.toString();
        try {
            byte[] data = Server.loadFixture("nonexistent", "");
            assertEquals(0, data.length);
        } finally {
            Server.FIXTURES_PATH = orig;
        }
    }
}

File: EggIncognito.CodeGen/Templates/kotlin/src/test/kotlin/ServerTest.kt
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.io.TempDir
import java.io.File
import java.nio.file.Path
import kotlin.test.assertEquals
import kotlin.test.assertTrue

class ServerTest {

    companion object {
        const val EID_PROTO = "MhJFSTAwMDAwMDAwMDAwMDAwMDE="
    }

    @Test
    fun extractEidReturnsCorrectEid() {
        assertEquals("EI0000000000000001", extractEid(EID_PROTO))
    }

    @Test
    fun extractEidEmptyInput() {
        assertEquals("", extractEid(""))
    }

    @Test
    fun loadFixtureDefault(@TempDir dir: Path) {
        val defDir = dir.resolve("default").toFile().also { it.mkdirs() }
        File(defDir, "test.binpb").writeBytes(byteArrayOf(0x01, 0x02))

        val orig = FIXTURES_PATH
        FIXTURES_PATH = dir.toString()
        try {
            val data = loadFixture("test", "")
            assertTrue(data.contentEquals(byteArrayOf(0x01, 0x02)))
        } finally {
            FIXTURES_PATH = orig
        }
    }

    @Test
    fun loadFixtureEidOverride(@TempDir dir: Path) {
        File(dir.resolve("default").toString()).mkdirs()
        File(dir.resolve("eids/EI0000000000000001").toString()).mkdirs()
        File(dir.resolve("default/test.binpb").toString()).writeBytes(byteArrayOf(0x01))
        File(dir.resolve("eids/EI0000000000000001/test.binpb").toString()).writeBytes(byteArrayOf(0x02))

        val orig = FIXTURES_PATH
        FIXTURES_PATH = dir.toString()
        try {
            val data = loadFixture("test", "EI0000000000000001")
            assertTrue(data.contentEquals(byteArrayOf(0x02)))
        } finally {
            FIXTURES_PATH = orig
        }
    }

    @Test
    fun loadFixtureMissing(@TempDir dir: Path) {
        val orig = FIXTURES_PATH
        FIXTURES_PATH = dir.toString()
        try {
            val data = loadFixture("nonexistent", "")
            assertEquals(0, data.size)
        } finally {
            FIXTURES_PATH = orig
        }
    }
}
